cablelabs/snaps-kubernetes
# Added functionality to inject SSH keys into the nodes to provision with k8s when they do not already exist.

Fixes #135

#### What does this PR do?
Adds functionality to inject ssh keys into nodes to provision when they do not already exist as when setting up the machines with snaps-boot. Also updated CI scripts to enable one to test this both ways.
#### Do you have any concerns with this PR?
no
#### How can the reviewer verify this PR?
Run CI with and without key injection (see new boolean CI config parameter 'inject_keys')
#### Any background context you want to provide?
I had removed the old key injection routines as they were not required due to the new snaps-boot workflow and we no longer could assume the nodes would have ssh password access.
#### Screenshots or logs (if appropriate)
#### Questions:
- Have you connected this PR to the issue it resolves?
Yes
- Does the documentation need an update?
no
- Does this add new Python dependencies?
no
- Have you added unit or functional tests for this PR?
I have updated the CI scripts to exercise both scenarios
- Does this patch update any configuration files?
no
